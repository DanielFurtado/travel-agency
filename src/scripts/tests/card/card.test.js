import React from 'react';
import { shallow } from 'enzyme';
import Card from '../../../components/card/card';

describe('Card', () => {
  const card = shallow(<Card />);
  
  it('renders properly', () => {
    expect(card).toMatchSnapshot();
  });
});