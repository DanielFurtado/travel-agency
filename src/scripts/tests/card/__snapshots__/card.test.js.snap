// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card"
>
        <div
                className="image-wrapper"
        >
                <i
                        className="fas fa-search-plus"
                />
                <img
                        alt={undefined}
                        className="card-image"
                        src={undefined}
                />
                <i
                        className="arrow-right"
                />
                <i
                        className="arrow-up"
                />
        </div>
        <div
                className="card-info-wrapper"
        >
                <div
                        className="card-title"
                />
                <div
                        className="card-description"
                />
                <CardPrice />
        </div>
        <BookNowButton
                id={undefined}
        />
</div>,
      "className": "card-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="image-wrapper"
>
            <i
                        className="fas fa-search-plus"
            />
            <img
                        alt={undefined}
                        className="card-image"
                        src={undefined}
            />
            <i
                        className="arrow-right"
            />
            <i
                        className="arrow-up"
            />
</div>,
          <div
            className="card-info-wrapper"
>
            <div
                        className="card-title"
            />
            <div
                        className="card-description"
            />
            <CardPrice />
</div>,
          <BookNowButton
            id={undefined}
/>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-search-plus"
/>,
              <img
                alt={undefined}
                className="card-image"
                src={undefined}
/>,
              <i
                className="arrow-right"
/>,
              <i
                className="arrow-up"
/>,
            ],
            "className": "image-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-search-plus",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": undefined,
                "className": "card-image",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "arrow-right",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "arrow-up",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card-title"
/>,
              <div
                className="card-description"
/>,
              <CardPrice />,
            ],
            "className": "card-info-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "card-title",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "card-description",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card"
>
          <div
                    className="image-wrapper"
          >
                    <i
                              className="fas fa-search-plus"
                    />
                    <img
                              alt={undefined}
                              className="card-image"
                              src={undefined}
                    />
                    <i
                              className="arrow-right"
                    />
                    <i
                              className="arrow-up"
                    />
          </div>
          <div
                    className="card-info-wrapper"
          >
                    <div
                              className="card-title"
                    />
                    <div
                              className="card-description"
                    />
                    <CardPrice />
          </div>
          <BookNowButton
                    id={undefined}
          />
</div>,
        "className": "card-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="image-wrapper"
>
              <i
                            className="fas fa-search-plus"
              />
              <img
                            alt={undefined}
                            className="card-image"
                            src={undefined}
              />
              <i
                            className="arrow-right"
              />
              <i
                            className="arrow-up"
              />
</div>,
            <div
              className="card-info-wrapper"
>
              <div
                            className="card-title"
              />
              <div
                            className="card-description"
              />
              <CardPrice />
</div>,
            <BookNowButton
              id={undefined}
/>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-search-plus"
/>,
                <img
                  alt={undefined}
                  className="card-image"
                  src={undefined}
/>,
                <i
                  className="arrow-right"
/>,
                <i
                  className="arrow-up"
/>,
              ],
              "className": "image-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-search-plus",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": undefined,
                  "className": "card-image",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "arrow-right",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "arrow-up",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="card-title"
/>,
                <div
                  className="card-description"
/>,
                <CardPrice />,
              ],
              "className": "card-info-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "card-title",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "card-description",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
